### PR TITLE
Feature upgrade admin fee

### PR DESCRIPTION
**Pull request by marco-sundsk**
This is a prerequisite feature upgrade for stable-swap pool feature.  
More specific, this admin_fee feature does following things:  
* using AdminFees structure to manage exchange_fee and referral_fee, let them can be changed dynamically by owner;
* Make exchange_fee and referral_fee to be based on swap fee, which is set by user when create pools;  
  that means, a pool with pool fee 25bps has a total fee 25bps now instead of 30bps (25+4+1) in the past;  
  now, reasonable exchange_fee and referral_fee are (1600, 400), which means 16% and 4% in total fee goes to exchange and referral;
* Make exchange_fee goes to contract's liquidity instead of owners in the past. Now owners can call owner method `remove_exchange_fee_liquidity` to explicit claim exchange fee to owner's inner account.

### CODE REVIEW

File: ref-exchange/src/lib.rs [hunk 1]
```diff
@@ -529,7 +541,7 @@ mod tests {
         // Exchange fees left in the pool as liquidity + 1m from transfer.
         assert_eq!(
             contract.get_pool_total_shares(0).0,
-            33337501041992301475 + 1_000_000
+            100012503125976904 + 1_000_000
```
Comment (referencedev) [inline ref-exchange/src/lib.rs:544]: may be it's better to change the exchange and referral fee in `setup_contract` to 1600 and 400 so these numbers continue matching?

File: ref-exchange/src/owner.rs [hunk 2]
```diff
@@ -72,6 +73,42 @@ impl Contract {
...
+    }
+
+    /// Remove exchange fee liqudity to owner's inner account.
+    /// without any storage and fee.
+    #[payable]
+    pub fn remove_exchange_fee_liquidity(&mut self, pool_id: u64, shares: U128, min_amounts: Vec<U128>) {
+        assert_one_yocto();
+        self.assert_owner();
+        self.assert_contract_running();
+        let ex_id = env::current_account_id();
```
Comment (referencedev) [inline ref-exchange/src/owner.rs:90]: contract_id?

File: ref-exchange/src/owner.rs [hunk 3]
```diff
@@ -72,6 +73,42 @@ impl Contract {
...
+        let mut pool = self.pools.get(pool_id).expect("ERR_NO_POOL");
+        let amounts = pool.remove_liquidity(
+            &ex_id,
+            shares.into(),
+            min_amounts
+                .into_iter()
+                .map(|amount| amount.into())
+                .collect(),
+            AdminFees::zero(),
+
```
Comment (referencedev) [inline ref-exchange/src/owner.rs:101]: extra new line

**Review by referencedev — APPROVED** [review]
Looks good

File: ref-exchange/src/lib.rs (hunk 1 above)
Comment (marco-sundsk) [reply ref-exchange/src/lib.rs:544]: good advice, change them in upgrade func maybe better.

**Pull request merged**